FlowCrypt/Core/Core.swift

```diff
@@ -128,21 +131,30 @@ final class Core: KeyDecrypter, CoreComposeMessageType {
         )
     }
 
-    func composeEmail(msg: SendableMsg, fmt: MsgFmt, pubKeys: [String]?) throws -> CoreRes.ComposeEmail {
-        let r = try call("composeEmail", jsonDict: [
-            "text": msg.text,
-            "to": msg.to,
-            "cc": msg.cc,
-            "bcc": msg.bcc,
-            "from": msg.from,
-            "subject": msg.subject,
-            "replyToMimeMsg": msg.replyToMimeMsg,
-            "atts": msg.atts.map { att in ["name": att.name, "type": att.type, "base64": att.base64] },
-            "format": fmt.rawValue,
-            "pubKeys": pubKeys,
-        ], data: nil)
-        // this call returned no useful json data, only bytes
-        return CoreRes.ComposeEmail(mimeEncoded: r.data)
+    func composeEmail(msg: SendableMsg, fmt: MsgFmt, pubKeys: [String]?) -> Future<CoreRes.ComposeEmail, Error> {
+        Future<CoreRes.ComposeEmail, Error> { [weak self] promise in
+            guard let self = self else { return }
```

Comment:
Here this call used to by synchronous but you've made it into a `Future`. Is this because the call is too slow? So you don't want it blocking the UI? Or other reason?

Reply:
Right, it blocks UI for showing the spinner while request is in progress